apps/antalmanac/src/stores/Schedules.ts

```diff
@@ -461,20 +461,23 @@ export class Schedules {
 
             // Get the course info for each course
             const courseInfoDict = new Map<string, { [sectionCode: string]: CourseInfo }>();
-            for (const [term, courseSet] of Object.entries(courseDict)) {
+
+            const websocRequests = Object.entries(courseDict).map(async ([term, courseSet]) => {
                 const sectionCodes = Array.from(courseSet);
-                // Code from ImportStudyList
                 const courseInfo = getCourseInfo(
                     await queryWebsoc({
                         term: term,
                         sectionCodes: sectionCodes.join(','),
                     })
                 );
                 courseInfoDict.set(term, courseInfo);
-            }
+            });
+
+            await Promise.all(websocRequests);
 
             // Map course info to courses and transform shortened schedule to normal schedule
             for (const shortCourseSchedule of saveState.schedules) {
+                console.log(shortCourseSchedule);
```

Comment:
Is this an accident?

Otherwise, code looks good. I haven't checked if it works though

Reply:
good catch, will remove